src/cudamatrix/cu-matrix.cc

```diff
@@ -1439,6 +1439,53 @@ void CuMatrixBase<Real>::DiffTanh(const CuMatrixBase<Real> &value,
   }
 }
 
+
+template<typename Real>
+void CuMatrixBase<Real>::ReLU(const CuMatrixBase<Real> &src) {
+  KALDI_ASSERT(SameDim(*this, src));
+#if HAVE_CUDA == 1
+  if (CuDevice::Instantiate().Enabled()) {
+    Timer tim;
+
+    dim3 dimBlock(CU2DBLOCK, CU2DBLOCK);
+    dim3 dimGrid(n_blocks(src.NumCols(), CU2DBLOCK), n_blocks(src.NumRows(), CU2DBLOCK));
+
+    cuda_tanh(dimGrid, dimBlock, this->data_, src.data_, this->Dim(), src.Stride());
```

Comment:
Thank you Dan for pointing out the mistake. I fixed it. And, as you mentioned I also have implemented as in nnet2 using ApplyFloor and ApplyHeaviside in the nnet-activation.h itself. I found the later version to be bit slower comparatively.